test/pod.persona.test.ts

```diff
@@ -0,0 +1,229 @@
+  const admin = opts?.overrideAdmin ? opts.overrideAdmin : expectedPod.admin;
+  jest.spyOn(fetchers, 'getPodFetchersByAddressOrEns').mockResolvedValueOnce({
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    Controller: { podAdmin: jest.fn().mockResolvedValueOnce(admin) },
+    Safe: {
+      address: expectedPod.safe,
+      nonce: jest.fn().mockResolvedValueOnce({
+        toNumber: jest.fn().mockImplementation(() => 5),
+      }),
+      getThreshold: jest.fn().mockResolvedValueOnce({
+        toNumber: jest.fn().mockImplementation(() => 10),
+      }),
+    },
+    podId: expectedPod.id,
+    Name: { name: expectedPod.name },
+  });
+}
+
+const provider = new ethers.providers.InfuraProvider('goerli', {
+  infura: '69ecf3b10bc24c6a972972666fe950c8',
+});
+
+beforeAll(async () => {
+  init({ provider, network: 5 });
+});
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+});
+
+test('Mint via admin persona throws if signer is not included', async () => {
+  mockGetPodFetchersByAddress();
+  const pod = await getPod(expectedPod.name);
+  pod.getPersonas = jest.fn().mockResolvedValue([{ type: 'admin', address: expectedPod.admin }]);
+  const personas = await pod.getPersonas(expectedPod.admin);
+  await expect(pod.callAsPersona(pod.mintMember, [expectedPod.admin], personas[0])).rejects.toThrow(
+    'Expected sender to be signer, but received',
+  );
+});
+
+test('Mint via admin persona calls the function as an admin', async () => {
+  mockGetPodFetchersByAddress();
+  const pod = await getPod(expectedPod.name);
+  pod.getPersonas = jest.fn().mockResolvedValue([{ type: 'admin', address: expectedPod.admin }]);
+  const personas = await pod.getPersonas(expectedPod.admin);
+
+  const mockSigner = {
+    getAddress: jest.fn().mockResolvedValueOnce(expectedPod.admin),
+  };
+
+  const mockMint = jest.fn();
+  jest.spyOn(utils, 'getMetropolisContract').mockReturnValueOnce({
+    mint: mockMint,
+  });
+
+  await pod.callAsPersona(pod.mintMember, [expectedPod.admin], personas[0], mockSigner);
+  // expect(mockMint).toHaveBeenCalled();
+  expect(mockMint).toHaveBeenCalledWith(
+    expectedPod.admin,
+    expectedPod.id,
+    ethers.constants.HashZero,
+  );
+});
+
+test('Mint via admin persona throws if the signer is not the admin', async () => {
+  mockGetPodFetchersByAddress();
+  const pod = await getPod(expectedPod.name);
+  pod.getPersonas = jest.fn().mockResolvedValue([{ type: 'admin', address: expectedPod.admin }]);
+  const personas = await pod.getPersonas(expectedPod.admin);
+
+  const mockSigner = {
+    getAddress: jest.fn().mockResolvedValueOnce(expectedPod.member),
+  };
+
+  const mockMint = jest.fn();
+  jest.spyOn(utils, 'getMetropolisContract').mockReturnValueOnce({
+    mint: mockMint,
+  });
+
+  await expect(
+    pod.callAsPersona(pod.mintMember, [expectedPod.admin], personas[0], mockSigner),
+  ).rejects.toThrow('Signer was not admin');
```

Comment:
❓ Where is this error actually thrown in the code? I don't see it in the PR, so I assume it's not part of the code changes?

Reply:
This is an error within the mint member function so that's already been set up!